test/integration/pubsub/test_pubsub_requests.rb

```diff
@@ -50,117 +62,123 @@ def some_topic_name
     @some_topic ||= new_topic_name.tap do |t|
       @client.create_topic(t)
     end
-    @some_topic
   end
 
   def some_subscription_name
     # create lazily to speed tests up
     @some_subscription ||= new_subscription_name.tap do |s|
       @client.create_subscription(s, some_topic_name)
     end
-    @some_subscription
   end
 
-  def test_create_topic
-    result = @client.create_topic(new_topic_name)
+  # retry_times attempts the given block up to tries times.
+  # A failed attempt to considered to be any point where the block yields
+  # and a Google::Apis::Error or StandardError is rescued.
+  def retry_times(tries, &block)
```

Comment:
If possible, it'd be nice to use https://github.com/fog/fog-core/blob/master/lib/fog/core/wait_for.rb instead of creating a oneoff function